### PR TITLE
Default runtime is the newest runtime

### PR DESCRIPTION
**Pull request by bryceitoc9**
## Description

Addendum to https://github.com/aws/aws-toolkit-vscode/pull/1059 : `getDefaultRuntime` was returning the oldest runtime instead of the newest. This reverses which version it pulls, which should be the newest.

## Motivation and Context
Don't recommend deprecated runtimes to users.

## Testing

Manual tests: Python and Node handlers give the latest runtime version.
## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the changelog using the script `npm run newChange`

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/lambda/models/samLambdaRuntime.ts [hunk 1]
```diff
@@ -78,12 +78,14 @@ export function getRuntimeFamily(langId: string): RuntimeFamily {
  */
 export function getDefaultRuntime(runtime: RuntimeFamily): string | undefined {
     switch (runtime) {
+        // note: compareSamLambdaRuntime sorts from least value to greatest, to alphabetically sort runtimes
+        // thus, pull the last value since that will represent the greatest (and thus, newest) version
```
Comment (justinmk3) [inline src/lambda/models/samLambdaRuntime.ts:82]: Is this worth a 2-line comment? Seems obvious. Or the same idea can be expressed in 1 line:

    // Sorts is ascending order, so last() returns the newest runtime.

Alternatively if `compareSamLambdaRuntime()` had a docstring on it this comment would not be needed.

**Comment by bryceitoc9** [comment]
> 
> 
> A problem like this might be a good case for having explicit defaultXxxRuntime consts, and tests.

Consts I could see, I'm not sure what a decent test would look like for this, though. It's pretty much: a test that doesn't have a const would basically use the same logic to validate the value, and a test that does use a const would either break every time we add a new runtime, which would probably mean that we should just be exporting the value we'd be updating anyway.

**Comment by awschristou** [comment]
> > A problem like this might be a good case for having explicit defaultXxxRuntime consts, and tests.
> 
> Consts I could see, I'm not sure what a decent test would look like for this, though. It's pretty much: a test that doesn't have a const would basically use the same logic to validate the value, and a test that does use a const would either break every time we add a new runtime, which would probably mean that we should just be exporting the value we'd be updating anyway.

If consts are used, tests aren't necessary. If you put tests around the current implementation, its almost deliberate that you'd have to update the tests whenever you add a new latest-n-greatest runtime, because that's the one you want this intelligent code picking up on.

**Comment by bryceitoc9** [comment]
>> ...
> 
> If consts are used, tests aren't necessary. If you put tests around the current implementation, its almost deliberate that you'd have to update the tests whenever you add a new latest-n-greatest runtime, because that's the one you want this intelligent code picking up on.

Yeah, that's what I'm getting at: the tests for this boil down to having a const somewhere, so it does make an argument for a const. What would you rather see: a const we have to keep up to date, or a function with either no tests or relatively worthless ones?

The other thing we could do is manually enforce keeping our sets ordered and pulling the first/last element from those (see discussion in this thread: https://github.com/aws/aws-toolkit-vscode/pull/1059#discussion_r417633599 )

(geez I wish Github threaded top level comments...)

File: src/lambda/models/samLambdaRuntime.ts [hunk 2]
```diff
@@ -74,17 +83,8 @@ export function getRuntimeFamily(langId: string): RuntimeFamily {
...
-        case RuntimeFamily.NodeJS:
-            return nodeJsRuntimes.sort(compareSamLambdaRuntime).first()
-        case RuntimeFamily.DotNetCore:
-            return dotNetRuntimes.sort(compareSamLambdaRuntime).first()
-        case RuntimeFamily.Python:
-            return pythonRuntimes.sort(compareSamLambdaRuntime).first()
-        default:
-            return undefined
-    }
+    return DEFAULT_RUNTIMES.get(runtime)
```
[the remark this answers is not in the document]
Comment (bryceitoc9) [reply src/lambda/models/samLambdaRuntime.ts:89]: It should just return undefined. Caller can handle it.

**Pull request merged**